pkg/liqo-controller-manager/offloadingStatus-controller/remoteconditions_management.go

```diff
@@ -0,0 +1,85 @@
+package offloadingstatuscontroller
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog"
+
+	offv1alpha1 "github.com/liqotech/liqo/apis/offloading/v1alpha1"
+	mapsv1alpha1 "github.com/liqotech/liqo/apis/virtualKubelet/v1alpha1"
+	liqoconst "github.com/liqotech/liqo/pkg/consts"
+	liqoutils "github.com/liqotech/liqo/pkg/utils"
+)
+
+// chooseRightRemoteCondition selects the right remote condition in according to the remote namespace
+// phase obtained by means of NamespaceMap.Status.CurrentMapping.
+func chooseRightRemoteCondition(phase mapsv1alpha1.MappingPhase) offv1alpha1.RemoteNamespaceCondition {
+	var remoteCondition offv1alpha1.RemoteNamespaceCondition
+	switch {
+	case phase == mapsv1alpha1.MappingAccepted:
+		remoteCondition = offv1alpha1.RemoteNamespaceCondition{
+			Type:    offv1alpha1.NamespaceReady,
+			Status:  corev1.ConditionTrue,
+			Reason:  "RemoteNamespaceCreated",
+			Message: "Namespace correctly offloaded on this cluster",
+		}
+	case phase == mapsv1alpha1.MappingCreationLoopBackOff:
+		remoteCondition = offv1alpha1.RemoteNamespaceCondition{
+			Type:    offv1alpha1.NamespaceReady,
+			Status:  corev1.ConditionFalse,
+			Reason:  "CreationLoopBackOff",
+			Message: "Some problems occurred during remote Namespace creation",
+		}
+	case phase == mapsv1alpha1.MappingTerminating:
+		remoteCondition = offv1alpha1.RemoteNamespaceCondition{
+			Type:    offv1alpha1.NamespaceReady,
+			Status:  corev1.ConditionFalse,
+			Reason:  "TerminatingNamespace",
+			Message: "The remote Namespace is requested to be deleted",
+		}
+	// If phase is not specified.
+	default:
+		remoteCondition = offv1alpha1.RemoteNamespaceCondition{
+			Type:    offv1alpha1.NamespaceOffloadingRequired,
+			Status:  corev1.ConditionFalse,
+			Reason:  "ClusterNotSelected",
+			Message: "You have not selected this cluster through ClusterSelector fields",
+		}
+	}
+	return remoteCondition
+}
+
+// assignClusterRemoteCondition sets the right remote namespace condition in according to the remote namespace phase
+// written in NamespaceMap.Status.CurrentMapping.
+// If phase==nil the remote namespace condition OffloadingRequired=False is set.
+func assignClusterRemoteCondition(noff *offv1alpha1.NamespaceOffloading, phase mapsv1alpha1.MappingPhase, clusterID string) {
+	if noff.Status.RemoteNamespacesConditions == nil {
+		noff.Status.RemoteNamespacesConditions = map[string]offv1alpha1.RemoteNamespaceConditions{}
+	}
+
+	var remoteConditions []offv1alpha1.RemoteNamespaceCondition
```

Comment:
I suppose this should be initialized to the previous condition, if any.

Reply:
you are right, ty